cpp/src/arrow/compute/exec/util.h

```diff
@@ -92,7 +92,7 @@ class TempVectorStack {
   Status Init(MemoryPool* pool, int64_t size) {
     num_vectors_ = 0;
     top_ = 0;
-    buffer_size_ = size;
+    buffer_size_ = PaddedAllocationSize(size) + kPadding + 2 * sizeof(uint64_t);
```

Comment:
It's weird if you `Init` the TempVectorStack with one size and then it segfaults if you try to `alloc` that much memory. That's because `alloc` bumps the stack by `PaddedAllocationSize(size) + 2 * sizeof(uint64_t)`